pkg/cluster/bootstrap.go

```diff
@@ -46,14 +46,25 @@ func (c *Cluster) Bootstrap(ctx context.Context, snapshot bool) error {
 
 	if c.managedDB != nil {
 		if !snapshot {
+			isHTTP := c.config.JoinURL != "" && c.config.Token != ""
+			// For secondary servers, we attempt to connect and reconcile with the datastore.
+			// If that fails we fallback to the local etcd cluster start
+			if isInitialized && isHTTP && c.clientAccessInfo != nil {
```

Comment:
how is `c.ClientAccessInfo` non-nil here? I thought we didn't set it until down below, at
```golang
c.clientAccessInfo, _ = clientaccess.ParseAndValidateTokenForUser(c.config.JoinURL, c.config.Token, "server")
```

Reply:
We don't check the error though, so it can be nil.